addons/website_event_sale/tests/test_website_event_sale_pricelist.py

```diff
@@ -67,3 +68,34 @@ def test_pricelist_different_currency(self):
             self.WebsiteSaleController.pricelist(promo=None)
             self.so._cart_update(line_id=so_line.id, product_id=self.ticket.product_id.id, set_qty=1)
         self.assertEqual(so_line.price_reduce_taxexcl, 900, 'Incorrect amount based on the pricelist and its currency.')
+
+class TestWebsiteEventSaleorder(HttpCase, TestWebsiteEventSaleCommon):
+
+    def test_event_registration_sale_order(self):
+        """ Test saleorder is created for tickets having price only """
+        self.authenticate(None, None)
+        self.env['event.event.ticket'].create([{
+            'event_id': self.event.id,
+            'name': 'Free',
+            'product_id': self.product_event.id,
+            'price': 0,
+        }])
+        event_questions = self.event.question_ids
+        name_question = event_questions.filtered(lambda q: q.question_type == 'name')
+        email_question = event_questions.filtered(lambda q: q.question_type == 'email')
+        phone_question = event_questions.filtered(lambda q: q.question_type == 'phone')
+        self.url_open(f'/event/{self.event.id}/registration/confirm', data={
+            f'1-name-{name_question.id}': 'Bob',
+            f'1-email-{email_question.id}': 'bob@test.lan',
+            f'1-phone-{phone_question.id}': '8989898989',
+            '1-event_ticket_id': self.event.event_ticket_ids[0].id,
+            f'2-name-{name_question.id}': 'joe',
+            f'2-email-{email_question.id}': 'joe@test.lan',
+            f'2-phone-{phone_question.id}': '8989898988',
+            '2-event_ticket_id': self.event.event_ticket_ids[1].id,
+            'csrf_token': http.Request.csrf_token(self),
+        })
+        self.assertEqual(len(self.event.registration_ids), 2)  # Test attendees should be created for both registration
+        self.assertFalse(self.event.registration_ids[0].sale_order_id)  # Sale order should not be created for free tickets
+        self.assertTrue(self.event.registration_ids[1].sale_order_id)  # Sale order should be created for paid tickets
```

Comment:
Instead of a python comment, you can set that as the failure message of the assertion.

Only do it when useful though, stuff like `self.assertEqual(my_variable, 5, "my_variable should be 5")` is not necessary as the base message already provides that information.